mmv1/products/clouddomains/Registration.yaml

```diff
@@ -0,0 +1,522 @@
+        default_from_api: true
+        description: |
+          The desired renewal method for this Registration. The actual renewalMethod is automatically updated to reflect this choice.
+          If unset or equal to RENEWAL_METHOD_UNSPECIFIED, the actual renewalMethod is treated as if it were set to AUTOMATIC_RENEWAL.
+          You cannot use RENEWAL_DISABLED during resource creation, and you can update the renewal status only when the Registration
+          resource has state ACTIVE or SUSPENDED.
+
+          When preferredRenewalMethod is set to AUTOMATIC_RENEWAL, the actual renewalMethod can be set to RENEWAL_DISABLED in case of
+          problems with the billing account or reported domain abuse. In such cases, check the issues field on the Registration. After
+          the problem is resolved, the renewalMethod is automatically updated to preferredRenewalMethod in a few hours.
+      - !ruby/object:Api::Type::String
+        name: transferLockState
+        default_from_api: true
+        description: Controls whether the domain can be transferred to another registrar. Values are UNLOCKED or LOCKED.
+  - !ruby/object:Api::Type::NestedObject
+    name: dnsSettings
+    description: Settings controlling the DNS configuration of the Registration.
+    properties:
+      - !ruby/object:Api::Type::NestedObject
+        name: customDns
+        description: Configuration for an arbitrary DNS provider.
+        properties:
+          - !ruby/object:Api::Type::Array
+            name: nameServers
+            required: true
+            description: |
+              Required. A list of name servers that store the DNS zone for this domain. Each name server is a domain
+              name, with Unicode domain names expressed in Punycode format.
+            item_type: Api::Type::String
+          - !ruby/object:Api::Type::Array
+            name: dsRecords
+            description: |
+              The list of DS records for this domain, which are used to enable DNSSEC. The domain's DNS provider can provide
+              the values to set here. If this field is empty, DNSSEC is disabled.
+            item_type: !ruby/object:Api::Type::NestedObject
+              properties:
+                - !ruby/object:Api::Type::Integer
+                  name: keyTag
+                  description: The key tag of the record. Must be set in range 0 -- 65535.
+                - !ruby/object:Api::Type::String
+                  name: algorithm
+                  description: The algorithm used to generate the referenced DNSKEY.
+                - !ruby/object:Api::Type::String
+                  name: digestType
+                  description: The hash function used to generate the digest of the referenced DNSKEY.
+                - !ruby/object:Api::Type::String
+                  name: digest
+                  description: The digest generated from the referenced DNSKEY.
+      - !ruby/object:Api::Type::NestedObject
+        name: googleDomainsDns
```

Comment:
Looks like the field is deprecated. Do we still want to implement it?

Reply:
It is deprecated, but it was used in the internal guide that I was using to test the resource. If I used it to test, I figured it is used enough now to be included.